utils.lua

```diff
@@ -150,4 +150,16 @@ function nn.utils.addSingletonDimension(t, dim)
   return view
 end
 
+function nn.utils.contiguousView(output, input, ...)
+  output = output or input.new()
+  if input:isContiguous() then
+    output:view(input, ...)
+  else
+    output:resizeAs(input)
+    output:copy(input)
+    output:view(...)
```

Comment:
Check my second comment.
I believe you are not assigning to `output` the shape indicated from `...` (if only one tensor is provided as argument, the viewed tensor is returned and the original tensor is kept intact)